### PR TITLE
Removed extra apt packages, added build-args for Python, GCC, G++ versions

### PR DESCRIPTION
**Pull request by randerzander**
Saves a few mb of disk space and makes the build more configurable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,9 +41,11 @@ A Dockerfile is provided for building and installing LibGDF and PyGDF from their
...
 * Host's installed nvidia driver must support >= the specified CUDA version (9.2 by default).
-* Alternative CUDA_VERSION should be specified via Docker [build-arg](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg) in Dockerfile):) and as cudatoolkit version in conda_environments/gdf_build.yml
+* Alternative CUDA_VERSION should be specified via Docker [build-arg](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg) and as cudatoolkit version in conda_environments/gdf_build.yml
 * Alternate branches for libgdf and pygdf may be specified as Docker build-args LIBGDF_REPO and PYGDF_REPO. See Dockerfile for example.
 * Ubuntu 16.04 is the default OS for this container. Alternate OSes may be specified as Docker build-arg LINUX_VERSION. See list of [available images](https://hub.docker.com/r/nvidia/cuda/).
+* Python 3.6 is default, but other versions may be specified via PYTHON_VERSION build-arg
+* GCC & G++ 4.8 are default, but other versions may be specified via CC and CXX build-args respectively
```
Comment (kkraus14) [inline README.md]: If we're not building Arrow from source and are pulling arrow from conda then we're locked to GCC 4.8 for ABI compatibility. We should probably note that somewhere around here in the README.

**Comment by kkraus14** [comment]
@randerzander changes look good to me, anything else you want to add or change before merge?

**Comment by randerzander** [comment]
nope. This is everything.

**Comment by kkraus14** [comment]
Merging. Thanks!

**Pull request merged**